src/Score.php

```diff
@@ -69,66 +68,38 @@ class Score implements VersionableInterface, ComparableInterface
     /**
      * Constructor
      *
-     * @param float|array $aRawValue      the score value, may also be an array of properties
+     * @param float|array $aRawValue      the raw score value, may also be an array of properties
      * @param float       $aMin           the score minimum
      * @param float       $aMax           the score maximum
-     * @param float       $aScalingFactor the score scaling factor
+     * @param float       $aScaledValue   the scaled score
      */
-    public function __construct($aRawValue = null, $aMin = null, $aMax = null, $aScalingFactor = null) {
+    public function __construct($aRawValue = null, $aMin = null, $aMax = null, $aScaledValue = null) {
         if (!is_array($aRawValue)) {
             $aRawValue = [
                 'raw'    => $aRawValue,
                 'min'    => $aMin,
                 'max'    => $aMax,
-                'scaled' => $aScalingFactor
+                'scaled' => $aScaledValue
             ];
         }
         $this->_fromArray($aRawValue);
     }
 
     /**
-     * @param  float $aValue
-     * @throws InvalidArgumentException
-     * @return null
-     */
-    public function validate($aValue) {
-        if (!isset($this->min, $this->max)) {
-            return;
-        }
-        if ($aValue < $this->min || $aValue > $this->max) {
-            throw new InvalidArgumentException(
-                sprintf("Value must be between %s and %s", $this->min, $this->max)
-            );
-        }
-    }
-
-    /**
-     * @param  int $aPrecision a rounding precision integer
-     * @return null|float
-     */
-    public function getValue($aPrecision = self::DEFAULT_PRECISION) {
-        if (!isset($this->raw)) {
-            return null;
-        }
-        if (isset($this->scaled)) {
-            return round($this->raw * $this->scaled, $aPrecision);
-        }
-        return round($this->raw, $aPrecision);
-    }
-
-    /**
      * @param  float $value
      * @throws InvalidArgumentException
      * @return self
      */
     public function setScaled($value) {
-        if ($value < static::SCALE_MIN || $value > static::SCALE_MAX) {
-            throw new InvalidArgumentException(sprintf(
-                "Scale must be between %s and %s [%s]",
-                static::SCALE_MIN,
-                static::SCALE_MAX,
-                $value
-            ));
+        if ($value < static::SCALE_MIN) {
+            throw new InvalidArgumentException(
```

Comment:
`OutOfBoundsException` is suggested for invalid keys which aren't at play here. `OutOfRangeException` is indicated for compile time use cases, this isn't one. The notion of the latter makes sense in only one of these cases, that with scaled because we know the full range a pri ori, but I still think we are better off with consistent exception types to prevent the need for multiple catches.